src/main/java/net/levelz/stats/PlayerStatsManager.java

```diff
@@ -341,4 +341,25 @@ public static String getRandomSkillString(Random random) {
     public static void onLevelUp(PlayerEntity playerEntity, int playerLevel) {
     }
 
+    public float getLevelProgress(PlayerEntity playerEntity) {
+        if (!ConfigInit.CONFIG.useIndependentExp) {
+            return Math.min(getPlayerExp(playerEntity) / (float) this.getNextLevelExperience(), 1F);
+        }
+        return levelProgress;
+    }
+
+    public static int getPlayerExp(PlayerEntity playerEntity) {
+        int level = playerEntity.experienceLevel;
+        int exp = 0;
+        for (int i = 0; i < level; i++) {
+            if (i >= 30) {
+                exp += 112 + (i - 30) * 9;
+            } else {
+                exp += i >= 15 ? 37 + (i - 15) * 5 : 7 + i * 2;
```

Comment:
why such odd formula?

Reply:
This formula comes from "PlayerEntity.getNextLevelExperience()"
So if someone mixin this method, the calculation will be inaccurate. 
But there is no way, "PlayerEntity.addExperienceLevels()" does not modify the totalExperience, otherwise I can use the "PlayerEntity.totalExperience"

Reply:
But if someone mix in this method, it is fine cause your system is based on the vanilla xp

Reply:
I think so :)

Reply:
I mean the vanilla method

Reply:
If someone mixin the vanilla method, he will only be able to use independent experience.

Reply:
why? I would use the vanilla method and if someone mix in the mod, no problem

Reply:
If someone modifies the formula of the original calculation experience, unless it is changed synchronously here to make it compatible, it is actually unnecessary. After all, changing this formula has too much impact, and there must be many incompatibilities.